src/interp.rs

```diff
@@ -19,556 +19,148 @@
-        }
-
-        fn reset(&mut self) {
-            // TODO: Use slice::fill() once stabilized
-            for v in &mut *self.z {
-                *v = 0.0;
-            }
-            self.zi = 0;
-        }
-
-        fn process(&mut self, src: &[f32], dst: &mut [f32]) {
-            assert!(src.len().checked_mul(self.factor) == Some(dst.len()));
-            assert!(self.z.len() == self.delay * self.channels as usize);
-            assert!(self.filter.len() == self.factor);
-            assert!(self.zi < self.delay);
-
-            if src.is_empty() {
-                return;
-            }
-
-            let frames = src.len() / self.channels as usize;
-            let mut zi = self.zi;
-
-            for (src, (dst, z)) in src.chunks_exact(frames).zip(
-                dst.chunks_exact_mut(frames * self.factor)
-                    .zip(self.z.chunks_exact_mut(self.delay)),
-            ) {
-                zi = self.zi;
-
-                for (src, dst) in src.iter().zip(dst.chunks_exact_mut(self.factor)) {
-                    // Add sample to delay buffer
-                    //
-                    // TODO Ringbuffer without bounds checks for z/zi
-                    //
-                    // Safety: zi is checked to be between 0 and self.delay
-                    *unsafe { z.get_unchecked_mut(zi) } = *src;
-
-                    // Apply coefficients
-                    for (filter, dst) in self.filter.iter().zip(dst.iter_mut()) {
-                        let mut acc = 0.0;
-                        for (c, index) in &*filter.coeff {
-                            let i = (zi as i32 + *index as i32) % self.delay as i32;
-
-                            // Safety: zi is checked to be between 0 and self.delay
-                            acc += *unsafe { z.get_unchecked(i as usize) } as f64 * c;
-                        }
-
-                        *dst = acc as f32;
-                    }
-
-                    zi = (zi + self.delay - 1) % self.delay;
-                }
-            }
-
-            self.zi = zi;
-        }
-    }
+impl<Item> ArrayBuf<Item> for [Item; 24] {
+    const SIZE: usize = 24;
 }
 
-/// A trait to be generic over number of channels in a of frame
-///
-/// TODO: Might want to use dasp-frame instead here, but needs
-/// coordination with `Samples` trait
-trait Frame: Sized + Copy {
-    const CHANNELS: usize;
+impl<Item> ArrayBuf<Item> for [Item; 12] {
+    const SIZE: usize = 12;
+}
 
-    fn scale_add(&mut self, other: &Self, coeff: f32);
-    fn from_planar(slice: &[f32], stride: usize) -> Self;
+/// A circular buffer offering fixed-length continous views into data
+/// This is enabled by writing data twice, also to a "shadow"-buffer following the primary buffer,
+/// The tradeoff is writing all data twice, the gain is giving the compiler continuous view with
+/// predictable length into the data, unlocking some more optimizations
+#[derive(Clone, Debug)]
+struct RollingBuffer<A, T> {
+    buf: [T; TAPS],
```

Comment:
```
error[E0277]: arrays only have std trait implementations for lengths 0..=32
  --> src/interp.rs:47:5
   |
47 |     buf: [T; TAPS],
   |     ^^^^^^^^^^^^^^ the trait `std::array::LengthAtMost32` is not implemented for `[T; 48]`
   |
   = note: required because of the requirements on the impl of `std::fmt::Debug` for `[T; 48]`
   = note: required because of the requirements on the impl of `std::fmt::Debug` for `&[T; 48]`
   = note: required for the cast to the object type `dyn std::fmt::Debug`
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

With Rust 1.47.0. Needs beta or nightly, it seems.

Unclear why this built fine on the CI with stable.

Reply:
Oh, because I was building with 1.46. Nevermind! 1.47 actually works fine